lib/block_view_atomic_txns_test.go

```diff
@@ -0,0 +1,388 @@
+package lib
+
+import (
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"math"
+	"testing"
+)
+
+func TestDependentAtomicTransactionGeneration(t *testing.T) {
+	// Initialize test chain, miner, and testMeta.
+	testMeta := _setUpMinerAndTestMetaForAtomicTransactionTests(t)
+
+	// Initialize m0, m1, m2, m3, m4.
+	_setUpUsersForAtomicTransactionsTesting(testMeta)
+
+	// Generate 1,000 dependent atomic transactions.
+	atomicTxns := _generateDependentAtomicTransactions(testMeta, 100)
+
+	// Construct a new view to connect the transactions to.
+	utxoView, err := NewUtxoView(
+		testMeta.db, testMeta.params, testMeta.chain.postgres, testMeta.chain.snapshot, nil)
+	require.NoError(t, err)
+	blockHeight := testMeta.chain.BlockTip().Height + 1
+
+	// Connect the transactions to ensure they can actually be connected.
+	for _, txn := range atomicTxns {
+		// Connect the transaction.
+		txHash := txn.Hash()
+		_, _, _, _, err := utxoView.ConnectTransaction(
+			txn, txHash, blockHeight, 0, true, false)
+		require.NoError(t, err)
+	}
+
+	// Flush the view to ensure everything is working properly.
+	require.NoError(t, utxoView.FlushToDb(uint64(blockHeight)))
+
+	//
+	// Now we test that the transactions are truly dependent on each-other by reorganizing them.
+	//
+
+	// Reorganize the transactions.
+	initialTxn := atomicTxns[0]
+	atomicTxns[0] = atomicTxns[len(atomicTxns)-1]
+	atomicTxns[0] = initialTxn
+
+	// Initialize test chain, miner, and testMeta for failing use.
+	testMetaFail := _setUpMinerAndTestMetaForAtomicTransactionTests(t)
```

Comment:
Why do we need a new chain and miner here? We should only ever need one for a given test. Running multiple really messes up memory usage on our CI pods.